pipenv/cli/command.py

```diff
@@ -388,8 +388,12 @@ def shell(state, fancy=False, shell_args=None, anyway=False, quiet=False):
             sys.exit(1)
     # Load .env file.
     load_dot_env(state.project)
-    # Use fancy mode for Windows.
-    if os.name == "nt":
+    # Use fancy mode for Windows or pwsh on *nix.
+    if (
+        os.name == "nt"
+        or os.environ["PIPENV_SHELL"].split(os.path.sep)[-1] == "pwsh"
```

Comment:
I am not  sure if these lookups from os.environ should actually be gets in case they are not defined?